### DISCUSSION TITLE
Confusion around :StringSerializer

### DISCUSSION THREAD

**Discussion by p14n** [replikativ/datahike#660] [General]
Hi - 
I was interested in looking under the covers of datahike by using a filestore and a StringSerializer.  I first had to provide my own store so that I could expose the default serializer from my cfg:

```
;; use the filesystem as storage medium
(def cfg {:store {:backend :filestr :path "/tmp/example" :default-serializer :StringSerializer}})

...

(defmethod empty-store :filestr [{:keys [path default-serializer config]}]
  (fs/connect-fs-store path :opts {:sync? true :default-serializer default-serializer} :config config :default-serializer default-serializer))
```

but then realised that this fails because although the default store starts with 3 possible serializers, `datahike.index.persistent-set/di/add-konserve-handlers` replaces this map of 3 with only a FressianSerializer - meaning the serializer resolves to null.

Is this intentional? Is there any circumstance where the StringSerialzer can be used with Datahike or is that a konserve-only construct?

Cheers

**Comment by whilo** [comment]
Yes, we decided to opt for Fressian for now and provide all the handlers for it here https://github.com/replikativ/datahike/blob/main/src/datahike/index/persistent_set.cljc#L220. We could extend support to other serializers, I am not sure about the `StringSerializer` though, it is somewhat inefficient. Why are you opting for it?

**Reply by p14n** [reply]
Thanks @whilo - I was just digging into the internals and was interested - no requirement for support.

**Reply by whilo** [reply]
Ok, cool. Feel free to ask follow up questions here or on slack. I will close this for now.
